File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/weeks/persistence/entity/DayEntity.java
package com.tarbadev.witchcraft.weeks.persistence.entity;

import com.tarbadev.witchcraft.recipes.persistence.entity.RecipeEntity;
import com.tarbadev.witchcraft.weeks.domain.entity.Day;
import com.tarbadev.witchcraft.weeks.domain.entity.DayName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "days")
public class DayEntity {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Integer id;
  private String name;
  @ManyToOne
  @JoinColumn(name = "lunch_recipe_id")
  private RecipeEntity lunch;
  @ManyToOne
  @JoinColumn(name = "diner_recipe_id")
  private RecipeEntity diner;

  public DayEntity(Day day) {
    id = day.getId();
    name = day.getName().name();
    lunch = day.getLunch() != null ? new RecipeEntity(day.getLunch()) : null;
    diner = day.getDiner() != null ? new RecipeEntity(day.getDiner()) : null;
  }

  public Day day() {
    return Day.builder()
        .id(id)
        .name(DayName.valueOf(name))
        .lunch(lunch != null ? lunch.recipe() : null)
        .diner(diner != null ? diner.recipe() : null)
        .build();
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/weeks/persistence/entity/DayEntity.kt
package com.tarbadev.witchcraft.weeks.persistence.entity

import com.tarbadev.witchcraft.weeks.domain.entity.Day
import com.tarbadev.witchcraft.weeks.domain.entity.DayName
import javax.persistence.*

@Entity(name = "day")
data class DayEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Int = 0,
    private val name: String = "",
    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "day_id")
    private val meals: List<MealEntity> = ArrayList()
) {
  constructor(day: Day) : this(
      id = day.id,
      name = day.name.name,
      meals = day.meals.map { MealEntity(it) }
  )

  fun day(): Day {
    return Day(
        id = id,
        name = DayName.valueOf(name),
        meals = meals.map { it.meal() }
    )
  }
}
